### PR TITLE
Https issue 1032536

### PR DESCRIPTION
**Pull request by nastasi-oq**
To solve this problem we move from self-signed webserver certificate to self-signed Certification Authority certificate and  webserver certificate signed with the CA certificate.

To produce correct certificates you must create your own CA and then produce a certificate and a key as described on geonode site.

To fix another https issue we move from httplib2 0.7.2 to 0.7.4, the package is into oq-ui-api repo and authomatically upgraded during the installation.

### CODE REVIEW

**Comment by bwyss** [comment]
I am unable to get the https to function correctly on a test v-box from a clean installation. The https is enabled, but I the Oq platform applications do not render overly layers.
The documentation needs to be cleaned up. The process require a combination of information from two separate README files in two separate repositories.
I would also suggest to explicitly describe the required commands in the README files eg. "on the oq-platform new server run the oq-platform2https.sh script" should indicate that the http2https premarital is needed. 
Apache is reporting "[warn] RSA server certificate CommonName (CN) `<myMachineName>' does NOT match server name!?"

**Comment by nastasi-oq** [comment]
Trivial typo fixed.
All the documentation is cleaned up and enhanced.
The two separate README are required because are in 2 different scopes, public and private repos with 2 different goals.
The warning isn't a problem for development machines. For public machines with can set it to the real name.

File: bin/geonode-installation.sh [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
-# Version: v1.0.0
+# Version: v1.2.0 https-issue_1032536 branch
 # Guidelines
 #
 #    Configuration file manglings are done only if they not appear already made.
 #
 
+# TIPS
+#    to remove all:
+# apt-get remove --purge geonode ; rm -rf ./oq-ui-api/ oq-ui-client/ oq-ui-geoserver/ gem_tmp/ django-schemata/ /var/lib/openquake/* /etc/geonode
```
Comment (luigi-oq) [inline bin/geonode-installation.sh:12]: Could you extract this comment and create a geonode-purge.sh script?

File: bin/geonode-installation.sh [hunk 2]
```diff
@@ -411,6 +445,7 @@ git checkout $GEM_OQ_UI_API_GIT_VERS"
     cd /var/lib/geonode/
     source bin/activate
     cd src/GeoNodePy/geonode/
+    pip install --upgrade "$norm_dir/oq-ui-api/data/httplib2.pybundle"
```
Comment (luigi-oq) [inline bin/geonode-installation.sh:449]: I would notice the user about the installation of a package in the application python path.
E.g.  echo "Installing httplib2 from local bundle"

File: bin/geonode-installation.sh (hunk 1 above)
Comment (nastasi-oq) [inline bin/geonode-installation.sh:12]: Only when I or others will test it deeply.

File: bin/geonode-installation.sh (hunk 2 above)
Comment (nastasi-oq) [inline bin/geonode-installation.sh:449]: The previous version was installed from geonode bundle too.

We can add an echo that explains the upgrade anyway.

**Comment by bwyss** [comment]
Please update the grammatical errors on oq-private/ssl/README
If there is a required command like "sudo oq-ui-api/bin/oq-platform2https.sh http2https", I would suggest to indicate it in the oq-private/ssl/README instead of 'run the oq-platform2https.sh script'

otherwise LGTM

**Pull request merged**